dev/PagerControl/PagerControl.cpp

```diff
@@ -0,0 +1,773 @@
+                newMargin.Top = 0;
+                newMargin.Right = 0;
+                newMargin.Bottom = 0;
+                indicator.Margin(newMargin);
+                indicator.Width(element.ActualWidth());
+            }
+        }
+    }
+}
+
+void PagerControl::AppendButtonToNumberPanelList(const int pageNumber, const int numberOfPages)
+{
+    winrt::Button button;
+    button.Content(winrt::box_value(pageNumber));
+    const auto buttonClickedFunc = [this](auto const& sender, auto const& args) {
+        if (const auto button = sender.try_as<winrt::Button>())
+        {
+            const int unboxedValue = winrt::unbox_value<int>(button.Content());
+            SelectedPageIndex(unboxedValue - 1);
+        }
+    };
+    button.Click(buttonClickedFunc);
+    // Set the default style of buttons
+    button.Style(unbox_value<winrt::Style>(ResourceAccessor::ResourceLookup(*this, box_value(c_numberPanelButtonStyleName))));
+    winrt::AutomationProperties::SetName(button, ResourceAccessor::GetLocalizedStringResource(SR_PagerControlPageTextName) + L" " + winrt::to_hstring(pageNumber));
+    winrt::AutomationProperties::SetPositionInSet(button, pageNumber);
+    winrt::AutomationProperties::SetSizeOfSet(button, numberOfPages);
+    m_numberPanelElements.Append(button);
+}
+
+void PagerControl::AppendEllipsisIconToNumberPanelList()
+{
+    winrt::SymbolIcon ellipsisIcon;
+    ellipsisIcon.Symbol(winrt::Symbol::More);
+    m_numberPanelElements.Append(ellipsisIcon);
+}
+
+/* Interaction event listeners */
+void PagerControl::OnRootGridKeyDown(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args)
+{
+    if (args.Key() == winrt::VirtualKey::Left || args.Key() == winrt::VirtualKey::GamepadDPadLeft)
+    {
+        winrt::FocusManager::TryMoveFocus(winrt::FocusNavigationDirection::Left);
+    }
+    else if (args.Key() == winrt::VirtualKey::Right || args.Key() == winrt::VirtualKey::GamepadDPadRight)
+    {
+        winrt::FocusManager::TryMoveFocus(winrt::FocusNavigationDirection::Right);
+    }
+}
+
+void PagerControl::ComboBoxSelectionChanged(const winrt::IInspectable& sender, const winrt::SelectionChangedEventArgs& args)
+{
+    if (const auto comboBox = m_comboBox.get())
+    {
+        SelectedPageIndex(comboBox.SelectedIndex());
+    }
+}
+
+void PagerControl::NumberBoxValueChanged(const winrt::IInspectable& sender, const winrt::NumberBoxValueChangedEventArgs& args)
+{
+    SelectedPageIndex((int)(args.NewValue()) - 1);
+}
+
+void PagerControl::FirstButtonClicked(const IInspectable& sender, const winrt::RoutedEventArgs& e)
+{
+    SelectedPageIndex(0);
+    if (const auto command = FirstButtonCommand())
+    {
+        command.Execute(nullptr);
+    }
+}
+
+void PagerControl::PreviousButtonClicked(const IInspectable& sender, const winrt::RoutedEventArgs& e)
+{
+    // In this method, SelectedPageIndex is always greater than 1.
+    SelectedPageIndex(SelectedPageIndex() - 1);
```

Comment:
does this work if selected page index was already 0?

Reply:
In that case, the button is disabled and thus this method can't be invoked.